### PR TITLE
Add a bottom boundary layer example using CATKE

### PR DESCRIPTION
**Pull request by glwagner**

### CODE REVIEW

File: validation/vertical_mixing_closures/bottom_boundary_layer.jl [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+using Printf
+
+using Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities:
+    CATKEVerticalDiffusivity, MixingLength
+
+# Parameters
+Δz = 4          # Vertical resolution
+Lz = 256        # Extent of vertical domain
+Nz = Int(Lz/Δz) # Vertical resolution
+f₀ = 0.0        # Coriolis parameter (s⁻¹)
```
Comment (navidcy) [inline validation/vertical_mixing_closures/bottom_boundary_layer.jl:13]: Why so verbose to explicitly denote that coriolis parameter is zero?

File: validation/vertical_mixing_closures/bottom_boundary_layer.jl [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                                                      overwrite_existing = true)
+
+progress(sim) = @info string("Iter: ", iteration(sim), " t: ", prettytime(sim),
+                             ", max(u): ", maximum(model.velocities.u))
+simulation.callbacks[:progress] = Callback(progress, IterationInterval(100))
+
+@info "Running a simulation of $model..."
+
+run!(simulation)
+
```
Comment (navidcy) [inline validation/vertical_mixing_closures/bottom_boundary_layer.jl:66]: How about a plot?

**Review by navidcy — APPROVED** [review]
Made minor comments

File: validation/vertical_mixing_closures/bottom_boundary_layer.jl (hunk 1 above)
Comment (glwagner) [reply validation/vertical_mixing_closures/bottom_boundary_layer.jl:13]: To show users how it might be changed

**Comment by glwagner** [comment]
https://github.com/CliMA/Oceananigans.jl/assets/15271942/b41d7837-61bc-4da5-8b2c-6f2c740c307f

File: validation/vertical_mixing_closures/bottom_boundary_layer.jl (hunk 1 above)
Comment (navidcy) [reply validation/vertical_mixing_closures/bottom_boundary_layer.jl:13]: Good point!

**Comment by navidcy** [comment]
Great -- thanks!
The `e` looks a bit strange with 1 value very off the rest. Does that look OK to you?
(We don't need to deal with this in this PR -- I was just wondering.)

**Comment by francispoulin** [comment]
I also thought that the `e` plot looked odd.  Is it just one point or is better resolved than that?

**Pull request merged**